### PR TITLE
Support http check with body

### PR DESCRIPTION
**Pull request by xuebing1110**
Support http check with body

### CODE REVIEW

File: agent/structs/check_definition.go [hunk 1]
```diff
@@ -70,6 +71,7 @@ func (c *CheckDefinition) CheckType() *CheckType {
 		GRPCUseTLS:        c.GRPCUseTLS,
 		Header:            c.Header,
 		Method:            c.Method,
+		Body:              c.Body,
```
Comment (preetapan) [inline agent/structs/check_definition.go]: Add this new field to the documentation on checks here - https://github.com/hashicorp/consul/blob/master/website/source/docs/agent/checks.html.md 
and here - https://github.com/hashicorp/consul/blob/master/website/source/api/agent/check.html.md

Comment (xuebing1110) [reply agent/structs/check_definition.go]: I have added the new field to the documentation.

**Comment by pearkes** [comment]
Hey @xuebing1110 I think we still would appreciate hearing about your use case for this -- thank you for adding the body field, however!

**Comment by xuebing1110** [comment]
@preetapan @pearkes I use consul to check my restful service, for example, using `PUT` method to update something with body.

**Comment by preetapan** [comment]
@xuebing1110 Does your service not have a healthcheck endpoint that responds to GET requests with a 200 response code when healthy and 500 when unhealthy? That type of healthcheck usually does not need a request body, and Consul supports a common usecase like that.  It sounds like you are using a `PUT` request to make changes to the app as the healthcheck, generally healthchecks are meant to tell you whether the app is responding or not. 

If we allow configuring a request body, that becomes another potential DOS vector - for example configuring a check in the data directory of consul with a large payload as the body will cause the agent to not start up because it might run out of memory. We would need to add additional validation for allowed body size limits. All that adds more complexity which we prefer to avoid.

**Comment by pearkes** [comment]
Given we haven't heard anything based on our suggestions/questions above I'm going to close this issue, but I encourage you to comment and we can re-open it if you want to pick this up again.

Alternatively, if things have changed dramatically, feel free to create a new PR.

**Comment by jakubgs** [comment]
I find it weird this was closed. I've described my usecase here: https://github.com/hashicorp/consul/issues/4148#issuecomment-391373631
Some APIs(like HTTP RPC) accept only JSON POSTs, and the actions initiated that way do not modify anything.

**Comment by pearkes** [comment]
@jakubgs we're definitely still open to a PR here, but we closed it due to not hearing back from the original author. We're still tracking the issue as a good enhancement.

**Pull request closed without merging**